code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/pushtemplate/AEPPushPayload.kt

```diff
@@ -0,0 +1,248 @@
+/*
+  Copyright 2024 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.services.ui.pushtemplate
+
+import androidx.core.app.NotificationCompat
+import com.adobe.marketing.mobile.util.StringUtils
+import com.google.firebase.messaging.RemoteMessage
+import kotlin.jvm.Throws
+
+/**
+ * Class representing the data push payload received from Firebase.
+ *
+ * @param message [RemoteMessage] object received from the [com.google.firebase.messaging.FirebaseMessagingService]
+ */
+class AEPPushPayload(val message: RemoteMessage?) {
+    private lateinit var messageData: MutableMap<String, String>
+    private lateinit var messageId: String
+    private lateinit var deliveryId: String
```

Comment:
You can remove the getter methods with the following syntax
```suggestion
    lateinit var messageData: MutableMap<String, String>
        private set
    lateinit var messageId: String
        private set
    lateinit var deliveryId: String
        private set
```